hbase-procedure/src/main/java/org/apache/hadoop/hbase/procedure2/store/wal/WALProcedureStore.java

```diff
@@ -1070,7 +1072,13 @@ boolean rollWriter(long logId) throws IOException {
     long startPos = -1;
     newLogFile = getLogFilePath(logId);
     try {
-      newStream = CommonFSUtils.createForWal(fs, newLogFile, false);
+      FSDataOutputStreamBuilder<?, ?> builder = fs.createFile(newLogFile).overwrite(false);
+      if (builder instanceof DistributedFileSystem.HdfsDataOutputStreamBuilder) {
```

Comment:
We could add a replicate(FSDataOutputStreamBuilder) method in CommonFSUtils to abstract the instance of logic? Otherwise we need to write instanceof at difference place multiple times.

Reply:
My only concern is that we would then need to add hadoop-hdfs-client dependency to hbase-common, and I am not sure if that's a good idea.

Reply:
OK, could leave it as is for now.